### DISCUSSION TITLE
Is there a chance for recursive type support.

### DISCUSSION THREAD

**Discussion by moshilansm** [mrjono1/joi-to-typescript#361] [Q&A] [answered]
Joi support recursive types through the 'link' function.
see example here:
https://joi.dev/api/?v=17.9.1#linkref

but when i tried it with joi-to-typescript the type of the recursive child is resolved to unknown.

The definition of such recursive type is possible with typescript interfaces.
is there a change to support it?

**Comment by mrjono1** [comment, marked as answer]
I've thought about this and its a lot of parsing to get this to work, which I don't have the intention on working on.
But you can use the same meta({className:''}) on the target and on the ref tag that should work
